code/game/jobs/job/antag/xeno/xenomorph.dm

```diff
@@ -25,41 +25,77 @@
 
 	transform_to_xeno(H, XENO_HIVE_NORMAL)
 
-/datum/job/antag/xenos/proc/transform_to_xeno(mob/living/carbon/human/H, hive_index)
-	var/datum/mind/new_xeno = H.mind
+/datum/job/antag/xenos/proc/transform_to_xeno(mob/living/carbon/human/human_to_transform, hive_index)
+	var/datum/mind/new_xeno = human_to_transform.mind
 	new_xeno.setup_xeno_stats()
 	var/datum/hive_status/hive = GLOB.hive_datum[hive_index]
 
-	H.first_xeno = TRUE
-	H.set_stat(UNCONSCIOUS)
-	H.forceMove(get_turf(pick(GLOB.xeno_spawns)))
+	human_to_transform.first_xeno = TRUE
+	human_to_transform.set_stat(UNCONSCIOUS)
+	human_to_transform.forceMove(get_turf(pick(GLOB.xeno_spawns)))
 
 	var/list/survivor_types = list(
 		/datum/equipment_preset/survivor/scientist,
 		/datum/equipment_preset/survivor/doctor,
 		/datum/equipment_preset/survivor/security,
 		/datum/equipment_preset/survivor/engineer
 	)
-	arm_equipment(H, pick(survivor_types), FALSE, FALSE)
+	arm_equipment(human_to_transform, pick(survivor_types), FALSE, FALSE)
 
-	for(var/obj/item/device/radio/radio in H.contents_recursive())
+	for(var/obj/item/device/radio/radio in human_to_transform.contents_recursive())
 		radio.listening = FALSE
 
-	H.job = title
-	H.apply_damage(50, BRUTE)
-	H.spawned_corpse = TRUE
+	human_to_transform.job = title
+	human_to_transform.apply_damage(50, BRUTE)
+	human_to_transform.spawned_corpse = TRUE
 
-	var/obj/structure/bed/nest/start_nest = new /obj/structure/bed/nest(H.loc) //Create a new nest for the host
-	H.statistic_exempt = TRUE
-	H.buckled = start_nest
-	H.setDir(start_nest.dir)
-	H.update_canmove()
-	start_nest.buckled_mob = H
-	start_nest.afterbuckle(H)
+	//placing the nests on walls logic
+	var/count = 0
+	var/obj/structure/bed/nest/start_nest
+	var/list/turf/list_to_search = list(get_turf(human_to_transform))
+	while(isnull(start_nest))
+		count++
+		var/list/turf/new_entries = list()
+		for(var/turf/turf_to_search in list_to_search)
+			new_entries |=  list(get_step(turf_to_search, NORTH), get_step(turf_to_search, SOUTH), get_step(turf_to_search, EAST), get_step(turf_to_search, WEST))
+		var/list/turf/bad_entries = list()
+		for(var/turf/open/found_open_turf in new_entries)
+			for(var/obj/found_object in found_open_turf)
+				if(istype(found_object, /obj/structure/window/framed) || found_object.density)
+					bad_entries |= found_open_turf
+		for(var/turf/closed/wall/wall_in_range in new_entries)
+			var/list/turf/neighbor_turfs = list(get_step(wall_in_range, SOUTH), get_step(wall_in_range, EAST), get_step(wall_in_range, WEST))
+			test_name:
```

Comment:
this is pretty rare syntax (it is used like 4 times in the whole codebase) - you can just put this loop in a proc that returns when the later condition calling continue is true